src/math/integerhistogram1d.h

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#pragma once
+
+#include "math/data1d.h"
+#include "math/sampleddouble.h"
+#include <map>
+
+// Pure Integer Histogram
+class IntegerHistogram1D
+{
+    public:
+    IntegerHistogram1D();
+    ~IntegerHistogram1D() = default;
+    IntegerHistogram1D(const IntegerHistogram1D &source);
+    // Clear data
+    void clear();
+
+    /*
+     * Histogram Data
+     */
+    private:
+    // Minimum value for data (hard left-edge of first bin)
+    int minimum_;
+    // Maximum value for data (hard right-edge of last bin, adjusted to match bin width if necessary)
+    int maximum_;
```

Comment:
I'm inclined to keep some sort of limits in since we might want to restrict against binning "silly" values. I would suggest that we change them to `std::optional`s to make sure their intent is clear, but keep the zero as the minimum limit since preventing negative numbers seems sensible (but perhaps not always desired!).  I've changed the comments too.
```suggestion
    // Minimum value allowed for binning
    std::optional<int> minimum_{0};
    // Maximum value allowed for binning
    std::optional<int> maximum_;
```